src/esr.rs

```diff
@@ -7,58 +15,156 @@ pub struct Esr {
 pub enum Error {
     #[error("Length must be between 5 and 25.")]
     InvalidLength,
-    #[error("Number must start with 'RF'.")]
+    #[error("ESR requires only digits")]
     InvalidFormat,
     #[error("Checksum is invalid.")]
     InvalidChecksum,
+    #[error("The following IBAN is not a QR-IID")]
+    InvalidQriid,
+    #[error("Parsing error, this is a bug please report")]
+    ParseIntError(#[from] std::num::ParseIntError),
 }
 
 impl Esr {
+    #[deprecated(
+        since = "0.2.4",
+        note = "use `try_without_checksum() or try_with_checksum()` instead"
+    )]
     pub fn try_new(number: String) -> Result<Self, Error> {
         let number = number.replace(' ', "").trim_start_matches('0').to_string();
-        if number.len() > 27 {
+        if number.len() > ESR_MAX_LENGTH {
             return Err(Error::InvalidLength);
         }
         if !number.chars().all(char::is_numeric) {
             return Err(Error::InvalidFormat);
         }
+        is_checksum_valid(&number)?;
 
-        if number[number.len() - 1..number.len()]
-            != checksum((number[..number.len() - 1]).to_string())
-        {
-            return Err(Error::InvalidChecksum);
+        Ok(Self { number })
+    }
+
+    /// Instantiate a new [`Esr`] struct
```

Comment:
```suggestion
    /// Instantiate a new [`Esr`] struct
    
```

Add an empty line to actually separate the headline from the second line.